### PR TITLE
Added a jenkins job for python code cleanup

### PR DESCRIPTION
**Pull request by mraarif**
this PR intends to add a job to automate running cleanup scripts for python codebase like `isort`, `2to3`, `pyupgrade` and `black` etc.
Relevant JIRA issue [here](https://openedx.atlassian.net/browse/BOM-1771).

### CODE REVIEW

File: testeng/resources/cleanup-python-code.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+echo "Getting current sha..."
+cd "$REPO_NAME"
+export CURRENT_SHA=$(git rev-parse HEAD)
+
+echo "Running clean up scripts"
+black --line-length=120 .
+isort --recursive .
+2to3 -f asserts -f future -f except -f metaclass -w .
+
+find . -type f -name "*.py" | while read fname; do
```
Comment (mzulqarnain1) [inline testeng/resources/cleanup-python-code.sh:22]: `2to3` and `pyupgrade` both create backup files with extension `.bak`, I think after all the processing, all such files should be deleted.

Comment (mraarif) [reply testeng/resources/cleanup-python-code.sh:22]: done

**Comment by awais786** [comment]
@morenol plz review.

File: testeng/jobs/cleanupPythonCode.groovy [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+ parameters {
+        stringParam('repoName', null, 'URL of the target repository')
+        stringParam('pythonVersion', PYTHON_VERSION, 'Version of python to use.(optional)')
+        stringParam('packages', '', 'Comma separated list of packages to install')
+        stringParam('scripts', '', 'Comma separated list of scripts to run')
+    }
+
+      environmentVariables(
+            REPO_NAME: repoName,
```
Comment (jmbowman) [inline testeng/jobs/cleanupPythonCode.groovy:25]: Doesn't this need to be `'${repoName}'`?

File: testeng/jobs/cleanupPythonCode.groovy [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        multiscm {
+            git {
+                remote {
+                    credentials('jenkins-worker')
+                    url("git@github.com:edx/${repoName}.git")
+                }
+                branch("master")
+                extensions {
+                    cleanBeforeCheckout()
+                    relativeTargetDirectory(repoName)
```
Comment (jmbowman) [inline testeng/jobs/cleanupPythonCode.groovy:40]: Same question here.

File: testeng/jobs/cleanupPythonCode.groovy [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                extensions {
+                    cleanBeforeCheckout()
+                    relativeTargetDirectory('testeng-ci')
+                }
+            }
+        }
+
+          steps {
+           def packagesToInstall = packages.tokenize(",")
+           def scriptsToRun = scripts.tokenize(",")
```
Comment (jmbowman) [inline testeng/jobs/cleanupPythonCode.groovy:57]: I suspect you may need to do something differently here to access the parameters also.

**Comment by mraarif** [comment]
@jmbowman after few changes as you suggested, the script ran successfully with `manually-seed-one-job` (result [here](https://build.testeng.edx.org/job/manually-seed-one-job/651/console))

**Comment by mraarif** [comment]
@jmbowman you were right, I had to make a few changes to get this to pass in `manually-seed-one-job` (result [here](https://build.testeng.edx.org/job/manually-seed-one-job/652/console)), I didn't know we could verify jobs this way, thanks for the tip.

File: testeng/jobs/cleanupPythonCode.groovy [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+def githubTeamReviewers = ['arbi-bom']
+
+def emails = ['arbi-bom@edx.org']
+
+
+job('cleanup-python-code-job'){
+
+ parameters {
+        stringParam('repoName', null, 'URL of the target repository')
```
Comment (jmbowman) [inline testeng/jobs/cleanupPythonCode.groovy:18]: Shouldn't the help text say it's the name of the target repository?

Comment (mraarif) [reply testeng/jobs/cleanupPythonCode.groovy:18]: yes, it was based on my early understanding (to expect full repo URL from the user)

**Comment by mraarif** [comment]
here's a [PR](https://github.com/edx/super-csv/pull/35) generated by this job.

File: testeng/jobs/cleanupPythonCode.groovy [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                extensions {
+                    cleanBeforeCheckout()
+                    relativeTargetDirectory(repoName)
+                }
+            }
+            git {
+                remote {
+                    url('https://github.com/edx/testeng-ci.git')
+                }
+                branch('BOM-1771')
```
Comment (jmbowman) [inline testeng/jobs/cleanupPythonCode.groovy:52]: Remember to change this before merging.

**Pull request merged**